File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/android/src/main/kotlin/com/danhdue/{{name.snakeCase()}}/{{name.pascalCase()}}Plugin.kt
/*
 * Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.
 */

/*
 * coverage:ignore-file
 */

package com.danhdue.{{name.snakeCase()}}

import com.danhdue.{{name.snakeCase()}}.di.{{name.pascalCase()}}ComponentProvider
{{#has_ui}}
import com.danhdue.{{name.snakeCase()}}.platform.{{name.pascalCase()}}PlatformViewFactory
{{/has_ui}}
{{^has_ui}}
import com.danhdue.{{name.snakeCase()}}.platform.{{name.pascalCase()}}HostApi
import com.danhdue.{{name.snakeCase()}}.platform.{{name.pascalCase()}}HostApiImpl
{{/has_ui}}
import io.flutter.embedding.engine.plugins.FlutterPlugin

class {{name.pascalCase()}}Plugin : FlutterPlugin {
    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        val component = {{name.pascalCase()}}ComponentProvider.get(flutterPluginBinding.applicationContext)

{{#has_ui}}
        flutterPluginBinding.platformViewRegistry.registerViewFactory(
            VIEW_TYPE,
            {{name.pascalCase()}}PlatformViewFactory {
                component.get{{name.pascalCase()}}ViewModel()
            }
        )
{{/has_ui}}
{{^has_ui}}
        {{name.pascalCase()}}HostApi.setUp(
            flutterPluginBinding.binaryMessenger,
            {{name.pascalCase()}}HostApiImpl(component.getGetDataUseCase())
        )
{{/has_ui}}
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
{{^has_ui}}
        {{name.pascalCase()}}HostApi.setUp(binding.binaryMessenger, null)
{{/has_ui}}
    }

    companion object {
        const val VIEW_TYPE = "com.danhdue.{{name.snakeCase()}}/native_view"
    }
}

File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/ios/Classes/{{name.pascalCase()}}Plugin.swift
// Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.

// coverage:ignore-file

import Flutter
import UIKit

public class {{name.pascalCase()}}Plugin: NSObject, FlutterPlugin{{^has_ui}}, {{name.pascalCase()}}HostApi{{/has_ui}} {
    public static func register(with registrar: FlutterPluginRegistrar) {
{{#has_ui}}
        let viewModel = {{name.pascalCase()}}ViewModel()
        let factory = {{name.pascalCase()}}PlatformViewFactory(viewModel: viewModel)
        registrar.register(factory, withId: "com.danhdue.{{name.snakeCase()}}/native_view")
{{/has_ui}}
{{^has_ui}}
        let messenger = registrar.messenger()
        let api = {{name.pascalCase()}}Plugin()
        {{name.pascalCase()}}HostApiSetup.setUp(binaryMessenger: messenger, api: api)
{{/has_ui}}
    }
{{^has_ui}}

    public func getPlatformVersion() throws -> String {
        return "iOS " + UIDevice.current.systemVersion
    }
{{/has_ui}}
}
